src/siminterface/simulator.py

```diff
@@ -15,33 +15,44 @@
 
 
 class Simulator(SimulatorInterface):
-    def __init__(self, test_mode=False, test_dir=None):
+    def __init__(self,  network_file, service_functions_file, config_file, resource_functions_path="",
+                 test_mode=False, test_dir=None):
         # Number of time the simulator has run. Necessary to correctly calculate env run time of apply function
         self.run_times = int(1)
         self.test_mode = test_mode
         self.test_dir = test_dir
         # Create CSV writer
         self.writer = ResultWriter(self.test_mode, self.test_dir)
+        # init network, sfc, sf, and config files
+        self.network, self.ing_nodes = reader.read_network(network_file, node_cap=10, link_cap=10)
+        self.sfc_list = reader.get_sfc(service_functions_file)
+        self.sf_list = reader.get_sf(service_functions_file, resource_functions_path)
```

Comment:
This works if `resource_functions_path` defaults to an empty string?

Reply:
I don't know, but it should. I just moved the place of the line upwards, I didn't touch it otherwise.